tests/transition.md

```diff
@@ -277,3 +281,381 @@ All test cases described in this document are beginning in a pre-Merge world and
+  * `forkchoiceUpdated(headBlockHash: TB.blockHash, payloadAttributes: mergeTransitionBlockAttributes)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00..00, payloadId: null}`
+    * EL's head points to `INV_TB` if `INV_TB.TD < TTD`, and to `INV_TB.parent` if `INV_TB.parent.TD >= TTD`
+  
+  </details>
+
+* [ ] Transition on a valid chain
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- TB <- P1 <- P2 <- P3`, `TB` is a valid terminal block
+  * EL starts with `TB` as the head
+  * `newPayload(P1)`
+    * EL returns `{status: VALID, latestValidHash: payload.blockHash}`
+    * EL's head points to `TB`
+  * `forkchoiceUpdated(head: P1, safe: 0x00..00, finalized: 0x00..00)`
+    * EL returns `{status: VALID, latestValidHash: forkchoiceState.headBlockHash}`
+    * EL's head updated to `P1`
+    * `eth_getBlockByNumber(safe)` returns `-39001: Unknown block` error
+    * `eth_getBlockByNumber(finalized)` returns `-39001: Unknown block` error
+  * `newPayload(P2) + forkchoiceUpdated(head: P2, safe: P1, finalized: 0x00..00)`
+    * EL's head updated to `P2`
+    * `eth_getBlockByNumber(safe)` returns `-39001: Unknown block` error
+    * `eth_getBlockByNumber(finalized)` returns `-39001: Unknown block` error
+  * `newPayload(P3) + forkchoiceUpdated(head: P3, safe: P2, finalized: P1)`
+    * EL's head updated to `P3`
+    * `eth_getBlockByNumber(safe)` returns `P2`
+    * `eth_getBlockByNumber(finalized)` returns `P1`
+  
+  </details>
+
+* [ ] Transition on an invalid chain
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- INV_TB <- P1`, `INV_TB` is a *valid PoW* but an *invalid terminal* block:
+    * [ ] `INV_TB.TD < TTD`
+    * [ ] `INV_TB.parent.TD >= TTD` -- it might require a second EL with a higher `TTD` value
+  * EL starts with `INV_TB` as the head
+  * `newPayload(P1)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00..00}`
+    * EL's head points to `INV_TB` if `INV_TB.TD < TTD`, and to `INV_TB.parent` if `INV_TB.parent.TD >= TTD`
+
+* [ ] Re-org to chain with invalid transition block
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2`, `B: Genesis <- ... <- TB <- INV_P1`, `A.TB` and `B.TB` are valid terminal blocks
+  * EL starts with `A.TB` as the head
+  * `newPayload(A.P1) + newPayload(A.P2)`
+    * EL returns `{status: VALID, latestValidHash: payload.blockHash}`
+    * EL's head points to `A.TB`
+  * `forkchoiceUpdated(head: A.P2, safe: A.P1, finalized: 0x00..00)`
+    * EL's head updated to `A.P2`
+  * `newPayload(B.INV_P1)`
+    * EL returns `{status: INVALID/ACCEPTED, latestValidHash: null/0x00..00}`
+    * EL's head points to `A.P2`
+  * `forkchoiceUpdated(head: B.INV_P1, safe: 0x00..00, finalized: 0x00..00)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00..00}`
+    * EL's head points to `B.P1`
+  
+  </details>
+
+* [ ] Syncing with the chain having a valid transition
```

Comment:
This should be covered by https://github.com/ethereum/hive/blob/f2838a629dcd1a027a260cd10fc84899f89e3ba6/simulators/ethereum/engine/mergetests.go#L244 
Main difference being that `Bn` is not a direct parent of `TB`, and there are a lot of blocks in between them, but the syncing is verified.

Reply:
Then main thing to check here is `SYNCING` response and that client is eventually synced with the chain and passed the transition while was syncing. I think it's done by the test you have linked